source/distributed/full_distributed_tria.cc

```diff
@@ -0,0 +1,499 @@
+              auto cell_info = cell_infos[ref_counter].begin();
+              for (; cell_info != cell_infos[ref_counter].end(); ++cell_info)
+                {
+                  // find cell that has the correct
+                  while (cell_info->id != cell->id().template to_binary<dim>())
+                    cell++;
+
+                  // subdomain id
+                  if (cell->active())
+                    cell->set_subdomain_id(cell_info->subdomain_id);
+
+                  // level subdomain id
+                  if (settings & construct_multigrid_hierarchy)
+                    cell->set_level_subdomain_id(cell_info->level_subdomain_id);
+
+                  // boundary ids
+                  for (auto pair : cell_info->boundary_ids)
+                    cell->face(pair.first)->set_boundary_id(pair.second);
+                }
+            }
+        }
+
+      update_number_cache();
+    }
+
+
+
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::create_triangulation(
+      const std::vector<Point<spacedim>> &vertices,
+      const std::vector<CellData<dim>> &  cells,
+      const SubCellData &                 subcelldata)
+    {
+      AssertThrow(false,
+                  ExcMessage(
+                    "Use the function reinit() to create the triangulation!"));
+      dealii::parallel::Triangulation<dim, spacedim>::create_triangulation(
+        vertices, cells, subcelldata);
+    }
+
+
+
+    template <int dim, int spacedim>
+    Triangulation<dim, spacedim>::Triangulation(MPI_Comm       mpi_communicator,
+                                                const Settings settings)
+      : parallel::DistributedTriangulationBase<dim, spacedim>(
+          mpi_communicator,
+          (settings & construct_multigrid_hierarchy) ?
+            static_cast<
+              typename dealii::Triangulation<dim, spacedim>::MeshSmoothing>(
+              dealii::Triangulation<dim>::none |
+              Triangulation<dim,
+                            spacedim>::limit_level_difference_at_vertices) :
+            static_cast<
+              typename dealii::Triangulation<dim, spacedim>::MeshSmoothing>(
+              dealii::Triangulation<dim>::none),
+          false)
+      , settings(settings)
+    {}
+
+
+
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::update_number_cache()
+    {
+      parallel::Triangulation<dim, spacedim>::update_number_cache();
+
+      if (settings & construct_multigrid_hierarchy)
+        parallel::Triangulation<dim, spacedim>::fill_level_ghost_owners();
+    }
+
+
+
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::execute_coarsening_and_refinement()
+    {
+      Assert(false, ExcNotImplemented());
+    }
+
+
+
+    template <int dim, int spacedim>
+    bool
+    Triangulation<dim, spacedim>::prepare_coarsening_and_refinement()
+    {
+      return dealii::Triangulation<dim, spacedim>::
```

Comment:
you say it is not supported, but this looks reasonable...

Reply:
I would like to insert here an assert (it does not make sense that this function is called until we do not support refinement). But this is not possible since it is called by `dealii::Triangulation<dim,spacedim>::execute_coarsening_and_refinement()`. I have added a warning to the code.